### PR TITLE
Cc ownership transfers

### PR DESCRIPTION
**Pull request by mattt-w**
Adding support for if a creator contracts ownership is transferred.

- [x] **originalOwner** - Which will store the address of the owner prior to burning/renouncing the contract
- [x] **transferState** - Tracks the state of the contracts transfers
0 - not transferred
1 - transferred to dead address with no artworks minted (FULL BURN)
2 - transferred to dead address with artworks minted (HALF BURN)
3 - transferred to another active address
- [x] **hasBeenTransferred** - Boolean, marks if the contract has been transferred or not

OriginalOwner attribute has not been added as the deployer/creator attribute already tracks this

### CODE REVIEW

File: src/mappings/v4-creator-contracts/creator-contract.ts [hunk 1]
```diff
@@ -541,6 +541,16 @@ export function handleBuyNowPurchased(event: BuyNowPurchased): void {
...
     creatorContractEntity.owner = event.params.newOwner
+    creatorContractEntity.hasBeenTransferred = true;
+    if (event.params.newOwner.equals(DEAD_ADDRESS) || event.params.newOwner.equals(ZERO_ADDRESS)) {
+        if (creatorContractEntity.totalNumOfEditions > ZERO) {
+            creatorContractEntity.transferState = BigInt.fromI32(2);
+        } else {
+            creatorContractEntity.transferState = BigInt.fromI32(1);
+        }
+    } else {
+        creatorContractEntity.transferState = BigInt.fromI32(3);
```
Comment (jamesmorgan) [inline src/mappings/v4-creator-contracts/creator-contract.ts:552]: Could move these to statics? similar to `KodaVersions.ts` ?

File: schema.graphql [hunk 2]
```diff
@@ -575,6 +575,14 @@ type CreatorContract @entity {
     # banned - flagged by the creator contract factory
     isHidden: Boolean!
 
+    #Transfer details
+    # 0: not transferred
+    # 1: transferred to dead address with no artworks minted
+    # 2: transferred to dead address with artworks minted
+    # 3: transferred to another active address
+    transferState: BigInt!
+    hasBeenTransferred: Boolean!
```
Comment (jamesmorgan) [inline schema.graphql:584]: Is this not the same as using `transferState=0`?

Comment (mattt-w) [reply schema.graphql]: Removed as we can just check against `transferstate=0` like suggested

File: src/mappings/v4-creator-contracts/creator-contract.ts (hunk 1 above)
Comment (mattt-w) [reply src/mappings/v4-creator-contracts/creator-contract.ts]: Moved to static consts in transferStates.ts

**Pull request merged**